benchflow-dsl/src/test/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/explorationvalues/IntValuesTest.scala

```diff
@@ -50,4 +50,92 @@ class IntValuesTest extends JUnitSuite {
 
   }
 
+  @Test def rangeStepAdditionUserExplorationSpaceTest(): Unit = {
+
+    val rangeStepYaml: String =
+      """
+        |range: [1,10]
+        |step: "+2"
+      """.stripMargin
+
+    val terminationCriteria = rangeStepYaml.parseYaml.convertTo[Try[IntValues]]
+
+    Assert.assertTrue(terminationCriteria.isSuccess)
+
+    terminationCriteria.get.values should contain allOf (1, 3, 5, 7, 9, 10)
+
+    val terminationCriteriaYaml = terminationCriteria.get.toYaml
+
+    Assert.assertTrue(terminationCriteriaYaml.prettyPrint.contains("values"))
+
+  }
+
+  @Test def rangeStepMultiplicationUserExplorationSpaceTest(): Unit = {
+
+    val rangeStepYaml: String =
+      """
+        |range: [1,10]
+        |step: "*2"
+      """.stripMargin
+
+    val terminationCriteria = rangeStepYaml.parseYaml.convertTo[Try[IntValues]]
+
+    Assert.assertTrue(terminationCriteria.isSuccess)
+
+    terminationCriteria.get.values should contain allOf (1, 2, 4, 8, 10)
+
+    val terminationCriteriaYaml = terminationCriteria.get.toYaml
+
+    Assert.assertTrue(terminationCriteriaYaml.prettyPrint.contains("values"))
+
+  }
+
+  @Test def rangeStepPowerUserExplorationSpaceTest(): Unit = {
+
+    val rangeStepYaml: String =
+      """
+        |range: [2,20]
+        |step: "^2"
+      """.stripMargin
+
+    val terminationCriteria = rangeStepYaml.parseYaml.convertTo[Try[IntValues]]
+
+    Assert.assertTrue(terminationCriteria.isSuccess)
+
+    terminationCriteria.get.values should contain allOf (2, 4, 16, 20)
+
+    val terminationCriteriaYaml = terminationCriteria.get.toYaml
+
+    Assert.assertTrue(terminationCriteriaYaml.prettyPrint.contains("values"))
+
+  }
+
+  @Test def rangeStepPowerIdentityElementUserExplorationSpaceTest(): Unit = {
+
+    val rangeStepYaml: String =
+      """
+        |range: [2,20]
+        |step: "^1"
+      """.stripMargin
+
+    val terminationCriteria = rangeStepYaml.parseYaml.convertTo[Try[IntValues]]
+
+    Assert.assertTrue(terminationCriteria.isFailure)
```

Comment:
we do report type of failure but in the test I think it is sufficient for now to just check that it actually failed as expected.